### PR TITLE
Added Dictionary using Trie Python code

### LINKED ISSUE #24: Dictionary using Trie
# Description

Implement a dictionary using a [trie](https://www.hackerearth.com/practice/data-structures/advanced-data-structures/trie-keyword-tree/tutorial/) in which the user enters a word, if it is present in the prebuilt dictionary print its meaning/definition. The user should also be able to include words of his/her own choice in the  #dictionar.

## Required prebuilt words
- hacktoberfest
- git
- open-source
- issue
- contribution

## Language
- [x] C++: @dwij2212 
- [x] Java
- [x] Python: @arnab031

## Type of issue

- [x] Feature
- [ ] Bug

The checklist will be updated by the maintainers/author of this issue according to the language in which snippets are added or are proposed to be added.

### PR DESCRIPTION
**Pull request by arnab031**
# Added Dictionary using Trie Python code

Fixes: #24

## Type of change

- [Y] New feature (non-breaking change which adds functionality)

# Checklist:

- [Y] My code follows the style guidelines of this project
- [Y] I have performed a self-review of my own code
- [Y] I have commented my code, particularly in hard-to-understand areas
- [Y] I have made corresponding changes to the documentation
- [Y] My changes generates no new warnings
- [Y] I have added tests that prove my fix is effective or that my feature works
- [Y] New and existing unit tests pass locally with my changes
- [Y] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by deepika-rana — CHANGES_REQUESTED** [review]
All requirements completed except including the prebuilt words mentioned.

**Comment by arnab031** [comment]
I have added prebuilt words mentioned. Please check it.

**Review by deepika-rana — APPROVED** [review]

File: snippets/python/dictionary_using_trie_python.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
+# Dictionary using Trie
+
+## Trie
+Trie is an efficient data retrieval structure which builds upon the properties of tree structure. Word search complexitites
+can be brought to an optimal limit which depends on key length.
+
+## Traversing the trie
+Main application of the trie is storing words in the structure. This trie can be traversed by first approaching the root node then traversing along the path on which our 
+letter and the letter within the node's branch matches. You can learn more about inserting and searching, and tries in general [here](https://www.geeksforgeeks.org/trie-insert-and-search/)
```
Comment (sakshamarora1) [inline snippets/python/dictionary_using_trie_python.md]: Add this stuff under the Description sub-heading like this:

## Description
### Trie
Trie is an efficient data retrieval structure which builds upon the properties of tree structure. Word search complexitites
can be brought to an optimal limit which depends on key length.

### Traversing the trie
Main application of the trie is storing words in the structure. This trie can be traversed by first approaching the root node then traversing along the path on which our 
letter and the letter within the node's branch matches. You can learn more about inserting and searching, and tries in general [here](https://www.geeksforgeeks.org/trie-insert-and-search/)

File: snippets/python/dictionary_using_trie_python.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+## Python Code
```
Comment (sakshamarora1) [inline snippets/python/dictionary_using_trie_python.md]:
```suggestion
## Code
```

File: snippets/python/dictionary_using_trie_python.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+TrieNode definition, Recursive functions for searching, inserting and a few helper functions are given in the following code snippet.
```
Comment (sakshamarora1) [inline snippets/python/dictionary_using_trie_python.md]: Move this to under the `Description` block

**Review by sakshamarora1 — CHANGES_REQUESTED** [review]
Looks good! Just a few changes.

**Review by kaustubhgupta — APPROVED** [review]

**Review by sakshamarora1 — APPROVED** [review]

**Pull request merged**